superset-frontend/src/views/CRUD/data/dataset/AddDataset/LeftPanel/index.tsx

```diff
@@ -211,6 +213,16 @@ export default function LeftPanel({
 
   const encodedSchema = schema ? encodeURIComponent(schema) : undefined;
 
+  useEffect(() => {
+    const currentUserSelectedDb = getItem(
+      LocalStorageKeys.db,
+      null,
+    ) as DatabaseObject;
+    if (currentUserSelectedDb) {
+      setDatabase(currentUserSelectedDb);
```

Comment:
I tried clicking the "Create Dataset" button after creating a database and wasn't able to get the database to auto-populate on my machine – is it working on yours?  Haven't looked deeply into it, but I think the problem might be that even though this effect calls `setDatabase` correctly to update the selected DB in the `dataset` state object of the parent `AddDataset` component, the selected DB isn't sent back down to the child `DatabaseSelector` component so it never updates.  To fix this, you may need to pass more of the database object (or the whole dataset object) through `LeftPanel` so you can send it to `DatabaseSelector`.

https://user-images.githubusercontent.com/13007381/211922005-11e9c9b0-d3ba-41d4-9991-8bb125acdd50.mov

Reply:
Thanks for pairing with me to fix this! Fixed in [`this commit`](https://github.com/apache/superset/pull/22610/commits/a72d05829371f6a375e165dd2d3739474587fa63).